recipes-openxt/openxt-measuredlaunch/openxt-measuredlaunch/seal-system

```diff
@@ -301,39 +271,42 @@ forward)
         pcr_forward[19]=":${pcr19}"
     fi
 
+    rm -f /boot/system/tpm/forward_pcr.lst
     for p in ${pcr_selection}; do
         pcr_params="${pcr_params} -p ${p}${pcr_forward[${p}]}"
+        if [ -n "${pcr_forward[${p}]}" ]; then
+            echo "${p}${pcr_forward[${p}]}" >> /boot/system/tpm/forward_pcr.lst
+        fi
     done
 
-    echo "$pcr_params" > /boot/system/tpm/forward_pcr.lst
+    rotate_sealed_keyfiles
 
-    if [ -e "${sealed_key}" ]; then
-        mv ${sealed_key} ${sealed_key}.old
-    fi
-    if [ -e "${sealed_key}.sha1" ]; then
-        mv ${sealed_key}.sha1 ${sealed_key}.sha1.old
-    fi
-    if [ -e "${sealed_key}.sha256" ]; then
-        mv ${sealed_key}.sha256 ${sealed_key}.sha256.old
+    if ! tpm_seal -a ${hashalg} ${pcr_params} "${config_key}" "${sealed_key}"; then
+        err "forward seal of key failed"
     fi
 
-    tpm_seal -a ${hashalg} ${pcr_params} ${config_key} ${sealed_key}
-    [ $? -eq 0 ] || err "forward seal of key failed"
-
     exit 0
 ;;
 unseal)
     for p in ${pcr_selection}; do
         pcr_params="${pcr_params} -p ${p}"
     done
 
-    [ -e "${sealed_key}.${hashalg}" ] || err "sealed key file missing"
-    [ "${tpm2}" -eq 0 ] && [ -e "${sealed_key}.pub.${hashalg}" ] ||
+    if [ ! -e "${sealed_key}.${hashalg}" ]; then
+        err "sealed key file missing"
+    fi
+
+    if [ "${tpm2}" -eq "0" -a ! -e "${sealed_key}.pub.${hashalg}" ]; then
```

Comment:
You mostly eliminated tpm2, but I would switch the tpm2 variable to tpm_ver - set it to "1" or "2" (or "1.2" or "2.0") and then check against that.
`if [ "${tpm_ver}" -eq "2" -a ! -e "${sealed_key}.pub.${hashalg}" ];` reads more intuitively to me.